packages/mobile/src/screens/search-screen-v2/SearchScreenV2.tsx

```diff
@@ -35,18 +36,33 @@ import {
 const { getV2SearchHistory } = searchSelectors
 const Stack = createNativeStackNavigator()
 
+const itemKindByCategory: Record<SearchCategory, Kind | null> = {
+  all: null,
+  users: Kind.USERS,
+  tracks: Kind.TRACKS,
+  playlists: Kind.COLLECTIONS,
+  albums: Kind.COLLECTIONS
+}
+
 export const SearchScreenV2 = () => {
   const [query] = useSearchQuery()
   const [category] = useSearchCategory()
   const [filters] = useSearchFilters()
-
   const history = useSelector(getV2SearchHistory)
-  const showRecentSearches = history.length > 0
+  const categoryKind: Kind | null = category
+    ? itemKindByCategory[category]
+    : null
+
+  const filteredSearchItems = useMemo(() => {
```

Comment:
Maybe we pass `filteredSearchItems` into `RecentSearches` so that we don't have to do the filtering twice

Reply:
yea this is a good thought. i'll update